### PR TITLE
TASK_51745_51759_51764_51765 : Add the possibility to delete a wallet

### PR DESCRIPTION
**Pull request by GouadriaHanen**

### CODE REVIEW

File: wallet-api/src/main/java/org/exoplatform/wallet/rest/WalletAccountREST.java [hunk 1]
```diff
@@ -165,7 +165,7 @@ public Response setInitializationStatus(@ApiParam(value = "wallet address", requ
     }
     try {
       accountService.setInitializationStatus(address,
-                                             WalletInitializationState.valueOf(status.toUpperCase()),
```
Comment (ahamdi) [inline wallet-api/src/main/java/org/exoplatform/wallet/rest/WalletAccountREST.java:168]: This endpoint is for administrator, normal users could just remove their personal wallets and should not be able to delete wallets of other uses.

File: wallet-services/src/main/java/org/exoplatform/wallet/service/WalletAccountServiceImpl.java [hunk 2]
```diff
@@ -517,25 +517,11 @@ public void setInitializationStatus(String address,
       throw new IllegalStateException(CAN_T_FIND_WALLET_ASSOCIATED_TO_ADDRESS + address);
     }
 
-    if (!isUserRewardingAdmin(currentUser)) {
```
Comment (ahamdi) [inline wallet-services/src/main/java/org/exoplatform/wallet/service/WalletAccountServiceImpl.java:520]: This code should not be removed but refactored to consider the case when a user wants to delete his own wallet and recreate a new one

File: wallet-webapps-admin/webpack.watch.js [hunk 3]
```diff
@@ -1,27 +0,0 @@
...
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- * Lesser General Public License for more details.
- * You should have received a copy of the GNU Lesser General Public License
- * along with this program; if not, write to the Free Software Foundation,
- * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
- */
-const path = require('path');
-const merge = require('webpack-merge');
```
Comment (ahamdi) [inline wallet-webapps-admin/webpack.watch.js]: why did you delete this file ?

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: wallet-api/src/main/java/org/exoplatform/wallet/rest/WalletAccountREST.java (hunk 1 above)
Comment (GouadriaHanen) [reply wallet-api/src/main/java/org/exoplatform/wallet/rest/WalletAccountREST.java:168]: this endpoint is used only to reject the wallet creation which is not the case anymore: the administration no longer has control over the state of the wallet.

Comment (GouadriaHanen) [reply wallet-api/src/main/java/org/exoplatform/wallet/rest/WalletAccountREST.java:168]: done!

File: wallet-services/src/main/java/org/exoplatform/wallet/service/WalletAccountServiceImpl.java (hunk 2 above)
Comment (GouadriaHanen) [reply wallet-services/src/main/java/org/exoplatform/wallet/service/WalletAccountServiceImpl.java:520]: done!

File: wallet-webapps-admin/webpack.watch.js (hunk 3 above)
Comment (GouadriaHanen) [reply wallet-webapps-admin/webpack.watch.js]: reverted!

File: wallet-services/src/main/java/org/exoplatform/wallet/service/WalletAccountServiceImpl.java [hunk 4]
```diff
@@ -517,25 +516,32 @@ public void setInitializationStatus(String address,
...
+        if (oldInitializationState == WalletState.INITIALIZED) {
+          throw new IllegalAccessException("Wallet was already marked as initialized, thus the status for address " + address
+                  + " can't change to status " + initializationState.name());
+        }
     }
     wallet.setInitializationState(initializationState.name());
     accountStorage.saveWallet(wallet, false);
-
+    if( initializationState == WalletState.DELETED){
+      LOG.warn( currentUser + "changed its wallet status with address "
```
Comment (ahamdi) [inline wallet-services/src/main/java/org/exoplatform/wallet/service/WalletAccountServiceImpl.java]: No need for this log

File: wallet-services/src/main/java/org/exoplatform/wallet/service/WalletAccountServiceImpl.java [hunk 5]
```diff
@@ -517,25 +516,32 @@ public void setInitializationStatus(String address,
...
+              + address + " to " + initializationState.name());
+      try {
+        getListenerService().broadcast(WALLET_DELETED_EVENT, wallet, currentUser);
+      } catch (Exception e) {
+        LOG.error("Error while braodcasting wallet {} state modification", wallet, currentUser);
+      }    }
```
Comment (ahamdi) [inline wallet-services/src/main/java/org/exoplatform/wallet/service/WalletAccountServiceImpl.java]: Code should be reformatted

**Review by ahamdi — CHANGES_REQUESTED** [review]

**Review by ahamdi — APPROVED** [review]
